File: Java/2785.java
import java.util.*;

class Solution {

    private boolean isVowel(Character c) {
        return c == 'a' || c == 'e' || c == 'o' || c == 'u' || c == 'i'
                || c == 'A' || c == 'E' || c == 'O' || c == 'U' || c == 'I';
    }

    public String sortVowels(String s) {
        List<Integer> index = new ArrayList<>();
        List<Character> allVowels = new ArrayList<>();
        char[] charS = s.toCharArray();
        for (int i = 0; i < charS.length; i++) {
            if (isVowel(charS[i])) {
                allVowels.add(charS[i]);
                index.add(i);
            }
        }
        Collections.sort(allVowels);
        for (int i = 0; i < index.size(); i++) {
            charS[index.get(i)] = allVowels.get(i);
        }
        return new String(charS);
    }
}

File: Python/2785.py
class Solution:
    def sortVowels(self, s: str) -> str:
        vowels = []
        for c in s:
            if c.lower() in "aeiou":
                vowels.append(c)

        if not vowels:
            return s

        vowels.sort()
        result = list(s)
        vowels_index = 0
        for i, c in enumerate(result):
            if c.lower() in "aeiou":
                result[i] = vowels[vowels_index]
                vowels_index += 1

        return "".join(result)
